### DISCUSSION TITLE
What is the recommended way of declaring schema in Python projects?

### DISCUSSION THREAD

**Discussion by tech-cobber** [geldata/gel#1864] [Q&A] [answered]
I wonder how to declare schema in python projects. As soon as you have EdgeQL with declarative SDL and syntax highlighter for it, there is an intuitive decision of having `.esdl` (or better `.edgeql`?) files with this content:
```
    module my_first_custom_module {
        type Movie {
         ...
        }
        type Person {
         ...
        }
    }
```
so it will look like this:
- schema/my_first_custom_module.esdl
- schema/my_second_custom_module.esdl
- schema/my_third_custom_module.esdl
And then reading from files and executing migrations from python code with edgedb-python.

Is this the expected way of working with EdgeDB? 

Perhaps this question is more about the usage of database interface libraries, but I thought you might have had a specific solution in mind when you were developing EdgeQL.

**Comment by kurtrottmann** [comment, marked as answer]
Hi, there is an RFC about migration with related information: https://github.com/edgedb/rfcs/blob/master/text/1000-migrations.rst

> The recommended way to define the EdgeDB schema is via SDL files that are committed into the repository of a project that uses EdgeDB. The canonical location for the schema files is the `dbschema/` directory in the project's root folder.
> 
> The `edgedb create-migration` command reads the files in the schema folder and creates the migration script with the help of the database server. Migration scripts are stored to the `dbschema/migrations/` directory and can be committed into the project's repository. Migration files contain valid EdgeQL text and can be modified by the user directly, for example, to insert data migration queries.

**Reply by tech-cobber** [reply]
Oh, thank you! I should take a look at this RFC, didn't know it exists. It's basically what I was looking for :)

**Comment by 1st1** [comment]
As @kurtrottmann suggested this is documented in the RFC-1000, but not yet clarified in the EdgeDB documentation or has a well-defined workflow built around it. We'll be focusing on this in the next few weeks.

**Reply by tech-cobber** [reply]
Good to hear, that would help a lot. Please left somewhere here link on the issue if you need hand with this
